File: miscellaneous/graph-bfs/20230416/graph_bfs.py
from typing import List


def graph_bfs(adj_list) -> List[int]:
    values = []
    queue = [0]
    seen = [0]

    while queue:
        current = queue.pop(0)
        values.append(current)

        for connection in adj_list[current]:
            if connection not in seen:
                seen.append(connection)
                queue.append(connection)

    return values

File: miscellaneous/graph-bfs/20230416/graph-bfs.php
<?php

declare(strict_types=1);

function graphBFs(array $adjList): array
{
    $values = [];
    $seen = [0];
    $queue = [0];

    while ($queue) {
        $current = array_shift($queue);
        $values[] = $current;

        foreach ($adjList[$current] as $connection) {
            if (!in_array($connection, $seen)) {
                $seen[] = $connection;
                $queue[] = $connection;
            }
        }
    }

    return $values;
}